### PR TITLE
TCOMP-1885 : make services explicitly serializable

### PR DESCRIPTION
**Pull request by clesaec**
https://jira.talendforge.org/browse/TCOMP-1885

### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant

### CODE REVIEW

File: component-api/src/main/java/org/talend/sdk/component/api/component/AfterVariables.java [hunk 1]
```diff
@@ -33,8 +33,7 @@
  */
```
Comment (undx) [inline component-api/src/main/java/org/talend/sdk/component/api/component/AfterVariables.java:33]: please remove anything not related to PR. (will do a cleanup later)

File: component-api/src/main/java/org/talend/sdk/component/api/record/Schema.java [hunk 2]
```diff
@@ -54,7 +54,7 @@ default Entry getEntry(final String name) {
     Map<String, String> getProps();
```
Comment (undx) [inline component-api/src/main/java/org/talend/sdk/component/api/record/Schema.java:54]: please remove anything not related to PR.

File: component-api/src/main/java/org/talend/sdk/component/api/service/http/Configurer.java [hunk 3]
```diff
@@ -81,6 +81,8 @@
 
         /**
          * Prevents the client to follow redirections.
+         * 
+         * @return connection.
          */
```
Comment (undx) [inline component-api/src/main/java/org/talend/sdk/component/api/service/http/Configurer.java:86]: please remove anything not related to PR.

File: component-api/src/main/java/org/talend/sdk/component/api/service/record/RecordService.java [hunk 4]
```diff
@@ -86,6 +86,7 @@ Record create(Schema schema, Record fallbackRecord, BiFunction<Schema.Entry, Rec
      * Visit a record with a custom visitor.
      *
      * @param visitor the visitor to use to browse the record.
+     * @param record record to visit.
```
Comment (undx) [inline component-api/src/main/java/org/talend/sdk/component/api/service/record/RecordService.java:89]: please remove anything not related to PR.

File: component-api/src/main/java/org/talend/sdk/component/api/service/BaseService.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+import org.talend.sdk.component.api.service.serialization.Serial;
+
+import lombok.Data;
+
+/**
+ * Classe base for services to be declared Serializable and use Component Manager serialization system.
+ */
+@Data
+public class BaseService implements Serializable {
```
Comment (undx) [inline component-api/src/main/java/org/talend/sdk/component/api/service/BaseService.java:29]: As far as I understand, we may break our connectors code base? Should we inherit from that class ?

Comment (undx) [reply component-api/src/main/java/org/talend/sdk/component/api/service/BaseService.java:29]: hum, read rest of code no breaks apparently... ;-)

**Review by undx — COMMENTED** [review]
Some minor comments and questions. 
As it's for fixing some code checkers warnings and not a bug fix, it will needs QA on all environments.

File: component-api/src/main/java/org/talend/sdk/component/api/service/BaseService.java (hunk 5 above)
Comment (clesaec) [reply component-api/src/main/java/org/talend/sdk/component/api/service/BaseService.java:29]: right, no breaks :-)

**Review by undx — DISMISSED** [review]
LGTM

**Review by undx — APPROVED** [review]
LGTM

**Pull request merged**